test-data/unit/check-typeddict.test

```diff
@@ -2031,6 +2031,23 @@ class DummyTypedDict(TypedDict):
 [builtins fixtures/dict.pyi]
 [typing fixtures/typing-typeddict.pyi]
 
+[case testTypedDictTypeNarrowingWithFinalKey]
+from typing import Final, Optional, TypedDict
+
+KEY_NAME: Final = "bar"
+class Foo(TypedDict):
+    bar: Optional[str]
+
+foo = Foo(bar="hello")
+if foo["bar"] is not None:
+    string1: str = foo["bar"]
```

Comment:
Maybe `reveal_type(foo["bar"])` and `reveal_type(foo[KEY_NAME])`? It is more verbose (which is good for testing).

Reply:
Thanks for your review!